core/src/main/scala/org/http4s/util/encoding/UriCodingUtils.scala

```diff
@@ -0,0 +1,253 @@
+package org.http4s.util.encoding
+
+import java.nio.charset.Charset
+import java.nio.{ByteBuffer, CharBuffer}
+
+import org.http4s.Uri
+import org.http4s.parser.RequestUriParser
+import org.http4s.util.encoding.tags._
+import org.parboiled2.CharPredicate
+
+import scala.annotation.tailrec
+import scala.collection.SeqLike
+import scala.util.Try
+
+
+/** Goodies for URI encoding */
+object UriCodingUtils {
+  // https://www.w3.org/TR/html5/forms.html#application/x-www-form-urlencoded-encoding-algorithm
+  def FormQueryParam = CharPredicate.Alpha ++ CharPredicate.Digit ++ "*-._"
+
+  private val Utf8 = Charset.forName("UTF-8")
+  private val HexUpperCaseChars = (0 until 16) map { i ⇒ Character.toUpperCase(Character.forDigit(i, 16)) }
+
+  def verifyAuthority(authority: String): Try[Uri.Authority] = new RequestUriParser(authority).Authority.run()
```

Comment:
Only because that's what parboiled2 seemed to want to return; not a fan of `Try` either though, and happy to see it changed.